src/logic/function/basic.lean

```diff
@@ -326,15 +326,19 @@ begin
   { simp [h, hg.ne] }
 end
 
-lemma comp_update {α' : Sort*} {β : Sort*} (f : α' → β) (g : α → α') (i : α) (v : α') :
-  f ∘ (update g i v) = update (f ∘ g) i (f v) :=
+lemma apply_update {ι : Sort*} [decidable_eq ι] {α β : ι → Sort*}
+  (f : Π i, α i → β i) (g : Π i, α i) (i : ι) (v : α i) (j : ι) :
+  f j (update g i v j) = update (λ k, f k (g k)) i (f i v) j :=
```

Comment:
Could this (and the next) be a simp lemma?  And if it were, would it streamline the proof of `comp_linear_map` in `linear_algebra.multilinear`?

Reply:
Lean failed to apply this lemma using `simp` because of some unification issues, so I used an explicit `have ... := ...`.